lib/board_selection.py

```diff
@@ -0,0 +1,28 @@
+from .tictactoe_board import TicTacToeBoard
+
+
+class BoardSelection:
+
+    valid_board_sizes = [3, 4]
+
+    def __init__(self, ui):
+        self.ui = ui
+
+    def select_board_size(self):
+        self.ui.display("Welcome to TIC TAC TOE!")
+        self.ui.display("First, select a board size.")
+        size = int(self.ui.get_user_input())
+        while size not in self.valid_board_sizes:
+            self.ui.display(
+                "The size of the board should be "
+                + self.board_sizes_for_invalid_message()
+                + " try again!"
+            )
+            size = int(self.ui.get_user_input())
+        return TicTacToeBoard(self.ui, size)
+
+    def board_sizes_for_invalid_message(self):
+        sizes = ""
+        for size in self.valid_board_sizes:
+            sizes = sizes + str(size) + ","
```

Comment:
Nit: this is going to print out a comma after the last size in the list:

```
The size of the board should be 3,4, try again!
```

There is a nice built-in function that you could use to inject delimiters between each value in a list: https://www.w3schools.com/python/ref_string_join.asp

Reply:
Done!